Lib/glyphsLib/builder/smart_components.py

```diff
@@ -0,0 +1,87 @@
+import pickle
+
+from fontTools.varLib.models import VariationModel, normalizeValue
+from fontTools.ttLib.tables._g_l_y_f import GlyphCoordinates
+
+from glyphsLib.classes import GSLayer
+
+AXIS_MIN = 1
+AXIS_MAX = 2
+
+
+def normalized_location(layer, base_layer):
+    loc = {}
+    for axis_name, current_value in layer.smartComponentPoleMapping.items():
+        base_value = base_layer.smartComponentPoleMapping[axis_name]
+        if current_value == base_value:
+            loc[axis_name] = 0.0
+        elif base_value == AXIS_MIN and current_value == AXIS_MAX:
+            loc[axis_name] = 1.0
+        elif base_value == AXIS_MAX and current_value == AXIS_MIN:
+            loc[axis_name] = -1.0
+        else:
+            raise ValueError(
+                f"Strange axis mapping for axis {axis_name} in smart layer {base_layer}"
+            )
+    return loc
```

Comment:
IIUC, this method means that the (normalized) master coordinates can only ever be -1, 0 or +1? Hm ok, if that's how it is.